### ISSUE TITLE
v8.5.1 for Windows now needs AppXSVC always running

### ISSUE THREAD

**Issue by bokeron2020** [notepad-plus-plus/notepad-plus-plus#13446]
I'm running W10 x64 & Notepad++ v.851

This new version notepad++.exe needs AppXSVC running always.
I understand AppXSVC is needed to install *.msix modern shell package... but I don't install it, and I see no other reason to require AppXSVC so I don't understand why now this service is a must.

Can someone please explain the reasoning behind this? Can this be handled in a different way so this service isn't absolutely necessary?

**Comment by GurliGebis** [comment]
It should only install the msix package on Windows 11, so there shouldn't be a need for that service on Windows 10.

On Windows 11, it is used for the new way of doing "Edit with Notepad++"

**Comment by bokeron2020** [comment]
> It should only install the msix package on Windows 11, so there shouldn't be a need for that service on Windows 10.
> 
> On Windows 11, it is used for the new way of doing "Edit with Notepad++"


So... might this be considered an undesired behaviour? If it's not used, it shouldn't be required.

I hope someone who can solve this could take a look.

**Comment by GurliGebis** [comment]
Can you describe how it tells you it is needed?
The code path for registering the msix only runs on Windows 11, so I'm curious on how you are getting any errors about it on Windows 10

**Comment by GurliGebis** [comment]
I wrote the implementation of it, so yep, I'm looking at it 😃

**Comment by bokeron2020** [comment]
> I wrote the implementation of it, so yep, I'm looking at it 😃

Thank you :-)

**Comment by GurliGebis** [comment]
My guess is that it is caused by the manifest being updated to support the Windows 11 context menu.
May I ask why you have disabled a core Windows service?

**Comment by bokeron2020** [comment]
> My guess is that it is caused by the manifest being updated to support the Windows 11 context menu. May I ask why you have disabled a core Windows service?

I always modify Windows. For my own use, because I achieve half the footprint, half memory use and slightly better system speed. And also prevents some rare vulnerabilities.
But also because, for some installs, I need to prevent users installing software while allowing some high-ish access level.

Regarding AppXSVC, I disable it because is just a "core service" due to Windows Store, which I block anyways.

**Comment by GurliGebis** [comment]
@bokeron2020 it is used for programs distributed using MSIX (which doesn't require the store) and (it seems) programs that has been built to support the new context menu.

@donho do we plan on supporting when people go around disabling random stuff? my stance is that if stuff is a core part of the operating system, and people disable it, and then run into problems, that is on them - we cannot support that. (Same goes for people who use NTLite to remove components from the OS)

**Comment by donho** [comment]
@GurliGebis 
> do we plan on supporting when people go around disabling random stuff? my stance is that if stuff is a core part of the operating system, and people disable it, and then run into problems, that is on them - we cannot support that. (Same goes for people who use NTLite to remove components from the OS)

Notepad++ should be able to be run under minimal environment. So yes, we should remove maximum dependency.

**Comment by GurliGebis** [comment]
@bokeron2020 can you try with this test version that has the manifest changed?
http://download.notepad-plus-plus.org/repository/MISC/nppShell.TEST13/

**Comment by GurliGebis** [comment]
@donho I have tested it on my machine, and it works with the AppXSVC service stopped.
Lets wait for @bokeron2020 to verify.

**Comment by bokeron2020** [comment]
> @bokeron2020 can you try with this test version that has the manifest changed? http://download.notepad-plus-plus.org/repository/MISC/nppShell.TEST13/

I have just tried. **It runs perfectly with AppXSVC service disabled**.

I've tried installing it with and without installing context menu and the program runs fine in both cases. When installed, the extension menu is also working as expected so I suppose there's some fallback still in place.

Just to add my opinion, I didn't see this as a question of suporting or not modified installs but as a design decision to make the software request just what it really needs, when it really needs it. Lean code. Added benefit, less opportunities to fail.

Thanks both @GurliGebis and @donho for taking the time to look at this.
